### DISCUSSION TITLE
Remove line break \\r

### DISCUSSION THREAD

**Discussion by 5011-5011** [nextgenhealthcare/connect#5454] [Q&A]
**Could someone tell me how to remove line breaks "\\r" from incoming Json message.**
![Captura de pantalla 2022-10-06 090242](https://user-images.githubusercontent.com/98954172/194307704-070888b7-ddd8-4fdc-a333-b83e7efb334f.png)

I have tried to do an input filter, or also from the data type configuration, but nothing has worked so far. I would appreciate any help. **Thanks.******

**Comment by pacmano1** [comment]
That is intentionally escaped.  the \r is the segment separator for hl7.  The extra \ escapes the first one. Why do you need to remove it?

If you are planning on using that field data as HL7 you would be calling serializers on it.

**Reply by 5011-5011** [reply]
I understand your point. What I am trying to do:
1. I receive Json with various data, one of which is an HL7 message as shown in the attached image above. In the destinations section I made a mapper where I need to replace or remove that line break, so that when mirth writes the file and sends it to the destination, that line break does not appear. Attached actions:
![Captura de pantalla 2022-10-06 091701](https://user-images.githubusercontent.com/98954172/194310819-4d7210ac-d97a-4bdd-99f7-5b950afdb36b.png)

The following image shows how I receive the text file, and it is correct, but I need to remove the line break highlighted in the image.

![Captura de pantalla 2022-10-06 091936](https://user-images.githubusercontent.com/98954172/194311026-e1f83ad0-8aac-4d2b-8b0b-74b1c51bb11e.png)

**Comment by jonbartels** [comment]
`\\\\r` seems to work.

The text has `\` so to get the regex to match `\` you have to escape it **in the regex** as `\\`. 

https://regex101.com/r/n72bb1/1

**Reply by 5011-5011** [reply]
From what I understood it should be like this?

![image](https://user-images.githubusercontent.com/98954172/194316772-e1f9c9e7-e6bf-4f3f-8b48-f46ad5da949c.png)


However it doesn't work for me, or maybe I'm still doing something wrong...

**Reply by jonbartels** [reply]
You probably need `\\r` without the `( )`.

**Reply by 5011-5011** [reply]
remove() however still doesn't work...  :(

**Reply by jonbartels** [reply]
:( 

Ok try this. 

1. Open https://regex101.com/r/LBVS5w/1
2. Add your message text as the "Test String"
3. Select "EcmaScript" on the left so that it uses the same parser as Mirth
4. Enter the regex you tried in "Regular Expression"
5. Enter the subsittion you tried in "Substitution"
6. Click "Save"
7. Post the link

That will help the people here see your experiment and help offer advice.

Many of us also hang out in Mirth Slack if you want to join the chatroom and get live help.

**Reply by 5011-5011** [reply]
https://regex101.com/r/pgf6WG/1
replace "\\r" with "\p" however there is still a "\" ...

**Reply by jonbartels** [reply]
https://regex101.com/r/pgf6WG/3 ?

**Reply by 5011-5011** [reply]
I really appreciate your help, but I solved it in another way! I'll show with an image in case someone finds it useful...    :D

![image](https://user-images.githubusercontent.com/98954172/194337880-659583fa-4e2d-48ca-8694-35fedfc94fde.png)

What I did was modify .toString() with .replace("\\r"," ") and remove the String Replacement section.

I am very grateful for your collaboration, and taking the time to answer. I would like to be able to be in Mirth's Slack, how should I do it?
Cheers

**Reply by jonbartels** [reply]
https://github.com/nextgenhealthcare/connect#useful-links has the Slack information

**Comment by pacmano1** [comment]
I am not clear on the ask.

You receive the JSON blob and are wanting to write that blob to disk with \\r removed?  Or you are wanting to write the HL7 value in the JSON to its own file?

Or you are expecting your screenshot to have the below when it starts writing the HL7 part of the JSON string?

```
MSH|...
PID|...
```

**Reply by 5011-5011** [reply]
I just wanted to get the text string without the line break, and I was able to do it. Thanks in the same way for wanting to help! Cheers :)
